app/ui/lib/TableActionButton.tsx

```diff
@@ -0,0 +1,29 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * Copyright Oxide Computer Company
+ */
+
+import { Link } from 'react-router-dom'
+
+import { Button, buttonStyle } from './Button'
+
+/**
+ * Used _outside_ of the `Table`, this returns either a link or a button that allows the user to take the designated action
+ */
+type TableActionButtonProps = { label: string } & (
+  | { linkTo?: never; onClick?: () => void }
+  | { linkTo: string; onClick?: never }
+)
+export const TableActionButton = ({ label, linkTo, onClick }: TableActionButtonProps) =>
+  linkTo ? (
+    <Link to={linkTo} className={buttonStyle({ size: 'sm' })}>
+      {label}
+    </Link>
+  ) : (
+    <Button onClick={onClick} size="sm" className="shrink-0">
+      {label}
+    </Button>
+  )
```

Comment:
I do kind of like this `ButtonOrLink` concept, but I feel like maybe it should be more general, so we could use it in other places. I almost want to be able to do something like this:

```tsx
AboveTable.LinkOrButton = makeLinkOrButtonComponent({ size: 'sm' })
```

on the other hand, once you have this concept of the component factory, you don't _really_ have to combine them into one component. If the point is to make clear in one spot what styling you're supposed to use regardless of whether it's a link or button, you can just have two components, as long as they're lightweight to define and intuitive to call. So, like I said above, think something like this would be ok:

```tsx
export const AboveTableButton = (props: ButtonProps) => (
  <Button size="sm" className="shrink-0" {...props} />
)
export const AboveTableLink = (props: LinkProps) => (
  <Link className={buttonStyle({ size: 'sm' })} {...props} />
)
```

though the props spreads always make me nervous.